shell/utils/router.js

```diff
@@ -25,3 +25,66 @@ export function queryParamsFor(current, qp, defaults = {}) {
 
   return query;
 }
+
+export function getClusterFromRoute(to) {
+  let cluster = to.params?.cluster;
+
+  if (!cluster) {
+    cluster = findMeta(to, 'cluster');
+  }
+
+  return cluster;
+}
+
+export function getProductFromRoute(to) {
+  let product = to.params?.product;
+
+  if ( !product ) {
+    const match = to.name?.match(/^c-cluster-([^-]+)/);
+
+    if ( match ) {
+      product = match[1];
+    }
+  }
```

Comment:
There's nothing that requires this to be true of routes (especially in extensions). It's probably fine but there's definitely baked in assumptions. Same for the other similar methods.

Reply:
This was all brought over from authenticate, i shuddered a bit too.